### ISSUE TITLE
Generating varargs for record value[N](T[N]) does not compile

### ISSUE THREAD

**Issue by rhart** [jOOQ/jOOQ#5775]
I have a postgres column of type `TEXT[]`.  The corresponding record that is generated does not compile because it is trying to override a method defined with `String[]` as `String...`.  In other words, the generated class has a method like 

```
@Override
public FooRecord value12(String... value) {
    setFooCol12(value);
    return this;
}
```

whereas the compiler expects

```
@Override
public FooRecord value12(String[] value) {
    setFooCol12(value);
    return this;
}
```

The offending generation code seems to be here https://github.com/jOOQ/jOOQ/blob/db502db1659c3101369249f00ab08d6ed60af325/jOOQ-codegen/src/main/java/org/jooq/util/JavaGenerator.java#L1119

Is there any particular reason to use varargs here?

**Comment by lukaseder** [comment]
Thank you very much for reporting.

The reason is convenience (see https://github.com/jOOQ/jOOQ/issues/4164). There is a warning, which should be suppressed by the existing generated `@SuppressWarnings` annotations.

With what compiler and compiler version did you get the error?

**Comment by rhart** [comment]
I'm using oracle 1.8.0_60 but have -Werror set, so it's not ignoring this compiler warning.  

I don't think this kind of warning can be suppressed, there is not a suppress warning token for this scenario AFAIK.  Even compiling without -Werror the warning is output by the compiler.

**Comment by lukaseder** [comment]
> but have -Werror set, so it's not ignoring this compiler warning.

Hmm, OK, I see. I suggest you turn that off at least for generated code... I'll adapt the issue title accordingly.

> I don't think this kind of warning can be suppressed, there is not a suppress warning token for this scenario AFAIK

You're right, I overlooked this. Thought the Eclipse compiler suppressed this warning with `@SuppressWarnings("all")`, but it doesn't.

Note, I just remembered, there had already been a discussion about this warning on the mailing list:
https://groups.google.com/d/msg/jooq-user/zlDXGePK7e8/1x9E2vExEAAJ

We've decided to add code generation flags to turn off varargs generation:
https://github.com/jOOQ/jOOQ/issues/5733

If you don't disagree, I'll close this issue here as a duplicate of #5733
